### ISSUE TITLE
segmentation fault mirroring http server folder with many files

### ISSUE THREAD

**Issue by gilcu3** [lavv17/lftp#474]
lftp version: 4.8.3

I'm getting 
```
“lftp” terminated by signal SIGSEGV (Address boundary error)
```
When doing:
```
#!/usr/bin/lftp -f
lcd localdir
open http://domain.com/remotedir/
mirror --verbose=3 -c --delete-first  --parallel=4 exampledir
```
The remote dir is in a http server, using h5ai for listing. For other folders in the same server with less files it works, but for a folder with many files, it gives the aforementioned error while "Getting directory contents".

**Comment by gilcu3** [comment]
Compiled current master branch with debug enabled, and used gdb to get:
```
Program received signal SIGSEGV, Segmentation fault.
0x00005555556b76fc in parse_html (
    buf=0x7ffff42fd6ad "<a href=\"..\">Parent Directory</a></td><td class=\"fb-d\"></td><td class=\"fb-s\"></td></tr><tr><td class=\"fb-i\"><img src=\"/_h5ai/public/images/fallback/folder.png\" alt=\"folder\"/></td><td class=\"fb-n\"><a h"..., buf_len=4242905, eof=true, list=..., 
    set=0x5555559d8160, all_links=0x0, prefix=0x7fffffffe1f0, base_href=0x7fffffffe1e0, lsopt=0x0, color=0) at HttpDir.cc:1018
1018	      str_with_tags=alloca_strdup(str);
```
Will try to fix myself, but help would be appreciated from anyone who already understands the code :)

**Comment by gilcu3** [comment]
Should be fixed in #477

**Comment by lavv17** [comment]
merged #477
